### PR TITLE
Bump github.com/buildpacks/pack to `v0.20.0`

### PR DESCRIPTION
**Pull request by aemengo**
Could we prioritize bumping pack within the codebase? Multiple users are [reporting an edge case](https://github.com/buildpacks/lifecycle/issues/435#issuecomment-859045100) using the pack plugin on Mac M1 machines. This has been addressed in the latest release: **v0.20.0**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -107,16 +108,17 @@ require (
...
-	google.golang.org/grpc v1.33.1
+	google.golang.org/api v0.44.0
+	google.golang.org/genproto v0.0.0-20210602131652-f16073e35f0c
+	google.golang.org/grpc v1.38.0
 	google.golang.org/protobuf v1.26.0
 	gopkg.in/go-playground/assert.v1 v1.2.1 // indirect
 	k8s.io/api v0.19.4
 	k8s.io/apimachinery v0.19.4
 	k8s.io/client-go v0.19.4
+	sigs.k8s.io/structured-merge-diff/v4 v4.0.2 // indirect
```
Comment (briancain) [inline go.mod:121]: I think this is unrelated? I didn't end up with this change when I bumped buildpack to `v0.20.0` 🤔 

```
brian@localghost:waypoint λ go get github.com/buildpacks/pack@v0.20.0                                                                                                                                ±[main]
go get: upgraded github.com/buildpacks/pack v0.18.1 => v0.20.0
go get: upgraded github.com/google/go-cmp v0.5.5 => v0.5.6
go get: upgraded golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d => v0.0.0-20210402161424-2e8d93401602
go get: upgraded google.golang.org/api v0.25.0 => v0.44.0
go get: upgraded google.golang.org/genproto v0.0.0-20201022181438-0ff5f38871d5 => v0.0.0-20210602131652-f16073e35f0c
go get: upgraded google.golang.org/grpc v1.33.1 => v1.38.0
```

File: go.mod [hunk 2]
```diff
@@ -5,14 +5,15 @@ go 1.13
 require (
 	github.com/Azure/azure-sdk-for-go v42.3.0+incompatible
 	github.com/Azure/go-autorest/autorest v0.10.2
+	github.com/Azure/go-autorest/autorest/adal v0.9.5 // indirect
```
Comment (briancain) [inline go.mod:8]: I think this is an unrelated change too.

**Review by briancain — COMMENTED** [review]
Hey there @aemengo , absolutely! I think there might be some unrelated package updates that were included in go.mod. Could you remove those? Otherwise looks good to me! I tested it with one of our pack plugin examples and seems to work ok. I don't have an M1 mac so I can't speak to if it fixes that 😅

File: go.mod (hunk 1 above)
Comment (aemengo) [reply go.mod:121]: Forgive me, but how exactly are you bumping? 😅 All I did was:

```shell
go get github.com/buildpacks/pack@v0.20.0
```

Comment (briancain) [reply go.mod:121]: I ran the same command against `main`:

> brian@localghost:waypoint λ go get github.com/buildpacks/pack@v0.20.0  

I'm not sure what's up, you might of had previous changes staged? No worries, I can always open a PR to fix this too if you can't get it to play nice! Let me know ❤️

Comment (aemengo) [reply go.mod]: It should be resolved in the latest commit. Please give it another look.

File: go.mod [hunk 3]
```diff
@@ -101,22 +101,24 @@ require (
 	github.com/skratchdot/open-golang v0.0.0-20200116055534-eef842397966
 	github.com/slack-go/slack v0.6.5
 	github.com/stretchr/testify v1.7.0
+	github.com/vdemeester/k8s-pkg-credentialprovider v1.18.1-0.20201019120933-f1d16962a4db // indirect
```
Comment (briancain) [inline go.mod:104]: Hey! I still see a bunch of k8s things in here 🙈 sorry to be a pain, I want to make sure your contribution is added 😅

File: go.mod [hunk 4]
```diff
@@ -101,22 +101,24 @@ require (
...
-	google.golang.org/grpc v1.33.1
+	google.golang.org/api v0.44.0
+	google.golang.org/genproto v0.0.0-20210602131652-f16073e35f0c
+	google.golang.org/grpc v1.38.0
 	google.golang.org/protobuf v1.26.0
 	gopkg.in/go-playground/assert.v1 v1.2.1 // indirect
 	k8s.io/api v0.19.4
 	k8s.io/apimachinery v0.19.4
 	k8s.io/client-go v0.19.4
+	k8s.io/code-generator v0.20.1 // indirect
```
Comment (briancain) [inline go.mod:121]: This was added too.

**Review by briancain — COMMENTED** [review]
I see a bunch of extra stuff still! I'm hoping if we remove those the extra go packages in `go.sum` will go away too.

**Comment by briancain** [comment]
Oh wait @aemengo - apologies! no, I think you're good! I'm confused by what buildpack is adding 🤔 but I think this is buildpack adding these not you now. I will take a look now, because I see the same packages being added when I try to make the change. One moment.

**Review by briancain — APPROVED** [review]
Thanks again! Sorry for the go package weirdness 😄

**Comment by aemengo** [comment]
> Thanks again! Sorry for the go package weirdness 😄

It's no problem at all! It took me much longer navigating these `go.mod` files than I'd like to admit. Thank you.
